circuit.py

```diff
@@ -104,46 +104,43 @@ def apply_transformation(self) -> None:
                 if isinstance(self.ops[index + 1], Measurement):
                     break
                 
-                self.commute_pi_over_four_rotation(index)
+                self.commute_rotation(index)
                 index += 1
 
 
-    def commute_pi_over_four_rotation(self, index: int) -> None:
+    def commute_rotation(self, index: int) -> None:
```

Comment:
I think we might want to leave the pi_over_four in the name because it only applies to pi_over_four rotations